pkg/operator/etcdendpointscontroller/etcdendpointscontroller.go

```diff
@@ -114,28 +113,27 @@ func (c *EtcdEndpointsController) syncConfigMap(ctx context.Context, recorder ev
 		klog.Warningf("required configmap %s/%s will be created because it was missing: %w", operatorclient.TargetNamespace, "etcd-endpoints", err)
 	}
 
-	// create endpoint addresses for each node
-	nodes, err := c.nodeLister.List(labels.Set{"node-role.kubernetes.io/master": ""}.AsSelector())
+	members, err := c.etcdClient.MemberList(ctx)
 	if err != nil {
-		return fmt.Errorf("unable to list expected etcd member nodes: %v", err)
+		return fmt.Errorf("failed to get member list: %w", err)
 	}
-	endpointAddresses := map[string]string{}
-	for _, node := range nodes {
-		var nodeInternalIP string
-		for _, nodeAddress := range node.Status.Addresses {
-			if nodeAddress.Type == corev1.NodeInternalIP {
-				nodeInternalIP = nodeAddress.Address
-				break
-			}
+
+	endpointAddresses := make(map[string]string, len(members))
+	// Create endpoint addresses for each member of the cluster.
+	for _, member := range members {
+		if member.Name == "etcd-bootstrap" {
+			continue
 		}
-		if len(nodeInternalIP) == 0 {
-			return fmt.Errorf("unable to determine internal ip address for node %s", node.Name)
+		// Use of PeerURL is expected here because it is a mandatory field, and it will mirror ClientURL.
+		ip, err := dnshelpers.GetIPFromAddress(member.PeerURLs[0])
+		if err != nil {
+			return err
 		}
-		endpointAddresses[base64.StdEncoding.WithPadding(base64.NoPadding).EncodeToString([]byte(nodeInternalIP))] = nodeInternalIP
+		endpointAddresses[fmt.Sprintf("%016x", member.ID)] = ip
```

Comment:
Nit: Just curious why this format is being chosen. This is an etcd member ID convention right, not some limitation of the configmap key length or format.

Reply:
member id is uint64 this conversion allows you to see it the same way the etcdctl printer does. But this is actually a bug and should be `16` vs `016`

Reply:
```
$ printf '%16x\n' 919655480020446617
cc345c95a5c2d99
```
```
+------------------+---------+--------------------------------------------+---------------------------+---------------------------+------------+
|        ID        | STATUS  |                    NAME                    |        PEER ADDRS         |       CLIENT ADDRS        | IS LEARNER |
+------------------+---------+--------------------------------------------+---------------------------+---------------------------+------------+
|  bf238e4c09e3045 | started |  ip-10-0-208-53.us-west-1.compute.internal |  https://10.0.208.53:2380 |  https://10.0.208.53:2379 |      false |
| 766f9b55c68c424f | started | ip-10-0-184-143.us-west-1.compute.internal | https://10.0.184.143:2380 | https://10.0.184.143:2379 |      false |
| d9af47ef64346aaa | started |  ip-10-0-146-57.us-west-1.compute.internal |  https://10.0.146.57:2380 |  https://10.0.146.57:2379 |      false |
+------------------+---------+--------------------------------------------+---------------------------+---------------------------+------------+
```

Reply:
fixed

Reply:
ref: https://github.com/etcd-io/etcd/blob/main/etcdctl/ctlv3/command/printer.go#L176